server/cmd/mmctl/commands/roles.go

```diff
@@ -48,16 +49,58 @@ var RolesMemberCmd = &cobra.Command{
 	RunE: withClient(rolesMemberCmdF),
 	Args: cobra.MinimumNArgs(1),
 }
+var RolesListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List available roles",
+	Example: `  $ mmctl roles list
+  $ mmctl roles list --json`,
+	RunE: withClient(rolesListCmdF),
+	Args: cobra.NoArgs,
+}
 
 func init() {
 	RolesCmd.AddCommand(
 		RolesSystemAdminCmd,
 		RolesMemberCmd,
+		RolesListCmd,
 	)
 
 	RootCmd.AddCommand(RolesCmd)
 }
 
+func rolesListCmdF(c client.Client, cmd *cobra.Command, args []string) error {
+	var errs *multierror.Error
+
+	roles, resp, err := c.GetAllRoles(context.TODO())
+	if err != nil {
+		errs = multierror.Append(errs, fmt.Errorf("failed to get roles: %w", err))
+	}
+
+	jsonOut := viper.GetBool("json")
+
+	if len(roles) == 0 {
+		if resp != nil && (resp.StatusCode == 401 || resp.StatusCode == 403) {
+			printer.PrintError("You don't have permissions to list roles")
+		} else {
+			printer.Print("No roles found.")
+		}
+		return errs.ErrorOrNil()
+	}
+
+	if jsonOut {
+		for _, role := range roles {
+			printer.Print(role)
+		}
+	} else {
+		printer.Print("Available roles:")
+		for _, role := range roles {
+			printer.Print(fmt.Sprintf("- %s", role.Name))
+		}
+	}
+
+	return errs.ErrorOrNil()
```

Comment:
This should be `nil` because so far we didn't handle any error which means we assume no errors. Also with my suggestions above, there is no need to use this anymore.
```suggestion
	return nil
```